types/renderprops-universal.d.ts

```diff
@@ -172,15 +174,22 @@ export interface TransitionProps<
    * @default {}
    */
   leave?: TLeave | ((item: TItem) => TLeave)
-  /**
-   * Callback when the animation comes to a still-stand
-   */
-  onRest?: (ds: DS) => void
-
   /**
    * Values that apply to elements that are neither entering nor leaving (you can use this to update present elements), or: item => values
    */
   update?: TUpdate | ((item: TItem) => TUpdate)
+  /**
+   * Called when an item's transition will begin
+   */
+  onStart?: (item: TItem, phase: TransitionPhase) => void
+  /**
+   * Called when an item's transition has come to a stand-still
+   */
+  onRest?: (item: TItem, phase: TransitionPhase, restValues: DS) => void
```

Comment:
Fixed the argument types of the `onStart`, `onRest`, and `onFrame` props in `TransitionProps`.